Settings.c

```diff
@@ -52,6 +52,19 @@ ShowFastbootCallback (
 
 STATIC
 EFI_STATUS
+AutoselectLastBootCallback (
+  MENU_ENTRY* This
+)
+{
+  SettingBoolSet("ui-autoselect-last-boot", !This->ToggleEnabled);
+  This->ToggleEnabled = SettingBoolGet("ui-autoselect-last-boot");
+  InvalidateActiveMenu();
+  MainMenuUpdateUi();
```

Comment:
you don't need to call MainMenuUpdateUi because this toggle doesn't affect what the main menu looks like.